docs/rpm_packages.md

```diff
@@ -3,3 +3,27 @@
 MicroShift RPM development packages are intended to be used internally at Red Hat to facilitate the development and QE processes, **never** to be distributed to customers. The packages are produced every time a corresponding OpenShift nightly build is successful.
 
 > For more information, see the `MicroShift Pre-Release RPM Packages` internal Red Hat Confluence page.
+
+## Building binaries from source RPM
+
+Following steps are applicable for source RPM from both release build system and developer's environment.
+
+> Hint: To build source RPM in developer environment run: `make srpm` - artifact will be in `_output/rpmbuild/SRPMS/`
+
+Assuming there's only one srpm in current directory, following commands will extract and enter the source directory:
+```bash
+rpm2cpio microshift-*.src.rpm  | cpio -idmv
+tar xf microshift-*.tar.gz
+cd microshift-*/
+```
+
+> Note: Source RPM does not include `.git/` directory.
+>
+> This means that commands (such as `make rpm`) expecting git repository will fail and 
+> version strings embedded in `microshift` binary are not fully populated.
+
+
+To build `microshift` and `microshift-etcd` binaries run:
+```bash
+make
+```
```

Comment:
Building a SRPM file on my system then moving it to a temporary directory and following these instructions gives me this error:

```
$ make
fatal: not a git repository (or any of the parent directories): .git
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
vendor/github.com/openshift/build-machinery-go/make/targets/golang/build.mk:15: *** no packages to build: GO_BUILD_PACKAGES_EXPANDED var is empty.  Stop.
```

It looks like something (the go compiler?) is still looking for a git repository.

Reply:
Interesting, for me it was successful. Do you run in on dev vm arm or on macos?

```
[microshift@ushift-default microshift-7dd54a861293b7bbd043921ab1ca1c28b8a31019]$ make
fatal: not a git repository (or any of the parent directories): .git
mkdir -p '_output/bin'
go build -mod=vendor -tags 'include_gcs include_oss containers_image_openpgp gssapi providerless netgo osusergo' -ldflags " -X k8s.io/component-base/version.gitMajor=1 -X k8s.io/component-base/version.gitMinor=26 -X k8s.io/component-base/version.gitVersion=v1.26.0 -X k8s.io/component-base/version.gitCommit=c7296782f65144a9aed5b18bfb880fd4f9b9333e -X k8s.io/component-base/version.gitTreeState=clean -X k8s.io/component-base/version.buildDate=2023-02-20T07:50:38Z -X k8s.io/client-go/pkg/version.gitMajor=1 -X k8s.io/client-go/pkg/version.gitMinor=26 -X k8s.io/client-go/pkg/version.gitVersion=v1.26.0 -X k8s.io/client-go/pkg/version.gitCommit=c7296782f65144a9aed5b18bfb880fd4f9b9333e -X k8s.io/client-go/pkg/version.gitTreeState=clean -X k8s.io/client-go/pkg/version.buildDate=2023-02-20T07:50:38Z -X github.com/openshift/microshift/pkg/version.majorFromGit=4 -X github.com/openshift/microshift/pkg/version.minorFromGit=13 -X github.com/openshift/microshift/pkg/version.versionFromGit=4.13.0-0.nightly-2023-02-15-202607-- -X github.com/openshift/microshift/pkg/version.commitFromGit= -X github.com/openshift/microshift/pkg/version.gitTreeState=clean -X github.com/openshift/microshift/pkg/version.buildDate=2023-02-20T07:50:38Z -w -s" -o '_output/bin/microshift' github.com/openshift/microshift/cmd/microshift
GO_LD_FLAGS=" -ldflags \"\
                   -X main.majorFromGit=4 \
                   -X main.minorFromGit=13 \
                   -X main.versionFromGit=4.13.0-0.nightly-2023-02-15-202607-- \
                   -X main.commitFromGit= \
                   -X main.gitTreeState=clean \
                   -X main.buildDate=2023-02-20T07:50:48Z \
				-w -s\"" \
	make -C etcd
make[1]: Entering directory '/home/microshift/srpms/microshift-7dd54a861293b7bbd043921ab1ca1c28b8a31019/etcd'
mkdir -p '../_output/bin'
go build -mod=vendor -tags 'include_gcs include_oss containers_image_openpgp gssapi providerless netgo osusergo' -ldflags " -X main.majorFromGit=4 -X main.minorFromGit=13 -X main.versionFromGit=4.13.0-0.nightly-2023-02-15-202607-- -X main.commitFromGit= -X main.gitTreeState=clean -X main.buildDate=2023-02-20T07:50:48Z -w -s" -o '../_output/bin/microshift-etcd' github.com/openshift/microshift/etcd/cmd/microshift-etcd
make[1]: Leaving directory '/home/microshift/srpms/microshift-7dd54a861293b7bbd043921ab1ca1c28b8a31019/etcd'
```

Reply:
I'm running in a VM. On that host, I can build MicroShift from the source tree (so I know I have the tools installed), but not from the SRPM.

Reply:
@pmtk, which version of go are you using? With 1.19.4, if I run `make GO_MOD_FLAGS='-buildvcs=false'` it seems to be working.

Reply:
I tried to reproduce this problem and everything seems to be OK. I'm using the following configuration
```
$ which go
/usr/local/bin/go
$ go version
go version go1.19.4 linux/amd64
$ cat /etc/redhat-release 
Red Hat Enterprise Linux release 8.7 (Ootpa)
```

Reply:
I also tried on RHEL9 and it seems to work, producing the following message:
```
fatal: not a git repository (or any of the parent directories): .git
```

Reply:
It seems to have something to do with where I unpack the tarball. The `-buildcvs` flag was added in go 1.18 (https://tip.golang.org/doc/go1.18#go-command) with the behavior of embedding version control information in the build automatically. If the compiler thinks it is in a git repo, it tries to include the details. ("Version control information is embedded if the go command is invoked in a directory within a Git, Mercurial, Fossil, or Bazaar repository, and the main package and its containing main module are in the same repository.") For some reason go thinks the directory in `~/tmp` meets those requirements, even though it is not version controlled. When I move to `/tmp` I no longer have the problem.

We may have fewer developer issues if our docs include the `GO_MOD_FLAGS` argument I mentioned above.

Reply:
In addition, we may want to use a full path in the command snippet, i.e. `~/microshift-srpm-builds`

Reply:
@dhellmann @ggiguash PTAL